qiskit/pulse/schedule.py

```diff
@@ -956,9 +1094,15 @@ def initialize_from(cls, other_program: Any, name: Optional[str] = None) -> "Sch
 
     @property
     def name(self) -> str:
-        """Name of this Schedule"""
+        """Return name of this schedule"""
         return self._name
 
+    @name.setter
+    def name(self, new_name: str):
+        """Update the name of this schedule."""
+        self._name = new_name
+        self._reference_manager._scope = new_name
+
```

Comment:
Is it important for the name to become mutable?  That feels like it might have knock-on effects.

Reply:
No. This is some artifact from initial implementation. Removed in https://github.com/Qiskit/qiskit-terra/pull/8005/commits/f4834569b9961dd7761e0326925b5a318ed8c07c